src/plugins/condition/Condition.js

```diff
@@ -211,35 +211,39 @@ export default class ConditionClass extends EventEmitter {
         }
     }
 
-    updateCriteriaResults(eventData) {
+    updateCriteriaResults(criteriaResults, eventData) {
         const id = eventData.id;
 
         if (this.findCriterion(id)) {
             // The !! here is important to convert undefined to false otherwise the criteriaResults won't get deleted when the criteria is destroyed
-            this.criteriaResults[id] = !!eventData.data.result;
+            criteriaResults[id] = !!eventData.data.result;
         }
     }
 
     handleCriterionResult(eventData) {
-        this.updateCriteriaResults(eventData);
+        this.updateCriteriaResults(this.criteriaResults, eventData);
         this.handleConditionUpdated(eventData.data);
     }
 
     requestLADConditionResult() {
-        const criteriaResults = this.criteria
+        let latestTimestamp;
+        let criteriaResults = {};
+        const criteriaRequests = this.criteria
             .map(criterion => criterion.requestLAD({telemetryObjects: this.conditionManager.telemetryObjects}));
 
-        return Promise.all(criteriaResults)
+        return Promise.all(criteriaRequests)
             .then(results => {
                 results.forEach(result => {
-                    this.updateCriteriaResults(result);
-                    this.latestTimestamp = this.getLatestTimestamp(this.latestTimestamp, result.data)
+                    this.updateCriteriaResults(criteriaResults, result)
+                    latestTimestamp = getLatestTimestamp(
+                        latestTimestamp,
+                        result.data,
+                        this.timeSystems
+                    );
                 });
-                this.evaluate();
-
                 return {
                     id: this.id,
-                    data: Object.assign({}, this.latestTimestamp, { result: this.result })
+                    data: Object.assign({}, latestTimestamp, { result: this.evaluate(criteriaResults) })
```

Comment:
You can also use object spread instead of Object.assign:
`{ ...this.latestTimestamp, ...{ result: this.result } }`

Reply:
Yeah... I should use spread more.